python/pylibcudf/pylibcudf/io/types.pyx

```diff
@@ -483,7 +482,7 @@ cdef class SourceInfo:
             for src in sources:
                 if not isinstance(src, (os.PathLike, str)):
                     raise ValueError("All sources must be of the same type!")
-                if not (os.path.isfile(src) or self._is_remote_file_pattern.match(src)):
+                if not (os.path.isfile(src) or SourceInfo._is_remote_uri(src)):
```

Comment:
```suggestion
                if not (os.path.isfile(src) or self._is_remote_uri(src)):
```
Nit

Reply:
I'm not sure if there's a particular preference. Generally using `self` would still automatically respect a subclass' override of `_is_remote_uri` which is a nice guarantee.

Probably not worth another commit & CI run if this PR is close to merging, but good to consider in the future.